bpf/bpf_lxc.c

```diff
@@ -973,19 +979,19 @@ static __always_inline int handle_ipv4_from_lxc(struct __ctx_buff *ctx, __u32 *d
 		 */
 		ep = lookup_ip4_endpoint(ip4);
 		if (ep) {
-#ifdef ENABLE_ROUTING
 			if (ep->flags & ENDPOINT_F_HOST) {
+				if (is_defined(ENABLE_ROUTING)) {
```

Comment:
Something that triggers me whenever I see this code - is the check for `ENDPOINT_F_HOST` actually needed? After we jump to `to_host`, we end up checking `*dst_id == HOST_ID` instead. Are the two conditions completely interchangeable?

If so we could also check for `HOST_ID` here, avoid the endpoint-lookup, and make the overall logic a bit simpler.